tool/clean/clean_log_group/clean_log_group.go

```diff
@@ -0,0 +1,280 @@
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatchlogs"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatchlogs/types"
+)
+
+// Config holds the application configuration
+type Config struct {
+	thresholdDays  int
+	inactiveDays   int
+	numWorkers     int
+	deleteBatchCap int
+	exceptionList  []string
+	dryRun         bool
+}
+
+// Logger wraps logging functionality
+type Logger struct {
+	*log.Logger
+}
+
+// NewLogger creates a new logger instance
+func NewLogger() *Logger {
+	return &Logger{
+		Logger: log.New(log.Writer(), "", log.LstdFlags),
+	}
+}
+
+// Global configuration
+var (
+	cfg    Config
+	logger *Logger
+)
+
+func init() {
+	// Initialize logger
+	logger = NewLogger()
+
+	// Set default configuration
+	cfg = Config{
+		thresholdDays:  3,
+		inactiveDays:   1,
+		numWorkers:     15,
+		deleteBatchCap: 10000,
+		exceptionList:  []string{"lambda"},
+	}
+
+	// Parse command line flags
+	flag.BoolVar(&cfg.dryRun, "dry-run", false, "Enable dry-run mode (no actual deletion)")
+	flag.Parse()
+}
+
+func main() {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Minute)
+	defer cancel()
+
+	// Load AWS configuration
+	awsCfg, err := loadAWSConfig(ctx)
+	if err != nil {
+		logger.Fatalf("Error loading AWS config: %v", err)
+	}
+
+	// Create CloudWatch Logs client
+	client := cloudwatchlogs.NewFromConfig(awsCfg)
+
+	// Compute cutoff times
+	cutoffTimes := calculateCutoffTimes()
+
+	logger.Printf("🔍 Searching for CloudWatch Log Groups older than %d days AND inactive for %d days in %s region\n",
+		cfg.thresholdDays, cfg.inactiveDays, awsCfg.Region)
+
+	// Delete old log groups
+	deletedGroups := deleteOldLogGroups(ctx, client, cutoffTimes)
+	logger.Printf("Total log groups deleted: %d", len(deletedGroups))
+}
+
+type cutoffTimes struct {
+	creation int64
+	inactive int64
+}
+
+func calculateCutoffTimes() cutoffTimes {
+	return cutoffTimes{
+		creation: time.Now().AddDate(0, 0, -cfg.thresholdDays).Unix() * 1000,
+		inactive: time.Now().AddDate(0, 0, -cfg.inactiveDays).Unix() * 1000,
+	}
+}
+
+func loadAWSConfig(ctx context.Context) (aws.Config, error) {
+	cfg, err := config.LoadDefaultConfig(ctx)
+	if err != nil {
+		return aws.Config{}, fmt.Errorf("loading AWS config: %w", err)
+	}
+	cfg.RetryMode = aws.RetryModeAdaptive
+	return cfg, nil
+}
+
+func deleteOldLogGroups(ctx context.Context, client *cloudwatchlogs.Client, times cutoffTimes) []string {
+	var (
+		wg                sync.WaitGroup
+		mutex             sync.Mutex
+		logGroupsToDelete []string
+		logGroupChan      = make(chan *types.LogGroup, 500)
+	)
+
+	// Start worker pool
+	logger.Printf("👷 Creating %d workers\n", cfg.numWorkers)
+	for i := 0; i < cfg.numWorkers; i++ {
+		wg.Add(1)
+		go processLogGroup(ctx, client, logGroupChan, &wg, &mutex, &logGroupsToDelete, times, i)
```

Comment:
nit: Consider making a worker struct that contains all of these fields. Could even have a single worker instance.

```go
type worker struct {
  client *cloudwatchlogs.Client
  logGroupChan <-chan *types.LogGroup
  ...
}

w := &worker{client: client, logGroupChan: logGroupChan, ...}

for i := 0; i < cfg.numWorkers; i++ {
  go w.processLogGroup(ctx, i)
}

if err := w.fetchAndProcessLogGroups(ctx); err != nil {
  ...
}
```
Would make it a bit easier to manage and understand.